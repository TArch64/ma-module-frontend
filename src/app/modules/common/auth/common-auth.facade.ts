import {Injectable} from "@angular/core";
import {CommonAuthService, User} from "@common/auth";
import {Observable} from "rxjs";

@Injectable()
export class CommonAuthFacade {
    constructor(
        private readonly authService: CommonAuthService,
    ) {}

    get isSignedOut(): boolean {
        return !this.isSignedIn;
    }

    get isSignedIn(): boolean {
        return this.authService.isSignedIn;
    }

    get authToken(): string | null {
        return this.authService.authToken;
    }

    get currentUser(): User | null {
        return this.authService.currentUser;
    }

    signOut(): void {
        this.authService.signOut();
    }

    fetchCurrentUser(): Observable<User> {
        return this.authService.fetchCurrentUser();
    }
}
