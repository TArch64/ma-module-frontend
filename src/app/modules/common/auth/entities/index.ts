export { User, IUserJSON } from './user';
export { InsecureUser, IInsecureUserJSON } from './insecure-user';
