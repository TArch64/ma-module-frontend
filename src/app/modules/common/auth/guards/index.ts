export {AuthOnlyGuard} from './auth-only.guard';
export {InauthOnlyGuard} from './inauth-only.guard';
