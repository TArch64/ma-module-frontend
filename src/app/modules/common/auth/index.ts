export {CommonAuthModule} from './common-auth.module';
export {CommonAuthService, ICommonAuthService} from './services';
export {AuthOnlyGuard, InauthOnlyGuard, RoleAccessGuard} from './guards';
export {AuthInterceptor} from './interceptors';
export {UserRoles} from './enums';
export {User, IUserJSON, InsecureUser, IInsecureUserJSON} from './entities';
