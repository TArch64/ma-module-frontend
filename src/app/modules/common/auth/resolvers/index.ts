export {ActualizeCurrentUserResolver} from './actualize-current-user.resolver';
