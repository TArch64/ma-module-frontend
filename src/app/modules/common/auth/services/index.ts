export {CommonAuthService} from './common-auth.service';
