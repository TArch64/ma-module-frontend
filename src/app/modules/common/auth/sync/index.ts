export { CommonAuthSyncService } from './common-auth-sync.service';
