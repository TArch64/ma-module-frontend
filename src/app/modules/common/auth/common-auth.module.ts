import {NgModule} from "@angular/core";
import {CommonModule} from "@angular/common";
import {HttpClientModule} from "@angular/common/http";
import {CommonAuthSyncService} from "./sync";
import {AuthOnlyGuard, InauthOnlyGuard, RoleAccessGuard} from "./guards";
import {AuthInterceptor} from "./interceptors";

@NgModule({
    imports: [
        CommonModule,
        HttpClientModule
    ],
    providers: [
        CommonAuthSyncService,
        AuthOnlyGuard,
        InauthOnlyGuard,
        AuthInterceptor,
        RoleAccessGuard
    ]
})
export class CommonAuthModule {}
