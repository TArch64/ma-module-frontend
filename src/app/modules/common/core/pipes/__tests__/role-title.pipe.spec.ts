import {UserRoles} from "@common/auth";
import {RoleTitlePipe} from "../role-title.pipe";

describe('transform', () => {
    const transform = (role: UserRoles) => new RoleTitlePipe().transform(role);

    test('should get admin title', () => {
        expect(transform(UserRoles.ADMIN)).toMatchSnapshot();
    });

    test('should get mentor title', () => {
        expect(transform(UserRoles.MENTOR)).toMatchSnapshot();
    });

    test('should get student title', () => {
        expect(transform(UserRoles.STUDENT)).toMatchSnapshot()
    });
});
