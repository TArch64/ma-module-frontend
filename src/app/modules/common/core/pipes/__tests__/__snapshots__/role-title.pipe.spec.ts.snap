// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform should get admin title 1`] = `"Admin Module"`;

exports[`transform should get mentor title 1`] = `"Mentor Module"`;

exports[`transform should get student title 1`] = `"Student Module"`;
